editoast/src/views/authz.rs

```diff
@@ -145,6 +152,123 @@ pub(in crate::views) async fn user_groups(
+        description = "A list of user IDs or identities to get information on",
+    ),
+    responses((
+        status = 200,
+        description = "Get information on a list of users",
+        body = inline(Vec<UserInfo>),
+    ))
+)]
+pub(in crate::views) async fn users_info(
+    Extension(auth): AuthenticationExt,
+    State(AppState {
+        regulator, db_pool, ..
+    }): State<AppState>,
+    Json(body): Json<Vec<UserIdOrIdentity>>,
+) -> Result<Json<Vec<UserInfo>>> {
+    let authorized = auth
+        .check_roles([Role::Admin].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+
+    // Retrieve all users ids and identity
+    // split ids and identities into two separate lists
+    let (user_ids, user_identities) = body.iter().fold(
+        (Vec::new(), Vec::new()),
+        |(mut ids, mut identities), id_or_identity| {
+            match id_or_identity {
+                UserIdOrIdentity::Id(id) => ids.push(*id),
+                UserIdOrIdentity::Identity(identity) => identities.push(identity.clone()),
+            }
+            (ids, identities)
+        },
+    );
+    // Retrieve users by ids
+    let settings = SelectionSettings::new().filter(move || User::ID.eq_any(user_ids.clone()));
+    let mut users = User::list(&mut db_pool.get().await?, settings).await?;
+    // Retrieve users by identities
+    let settings =
+        SelectionSettings::new().filter(move || User::IDENTITY_ID.eq_any(user_identities.clone()));
+    users.extend(User::list(&mut db_pool.get().await?, settings).await?);
+
+    // Fetch groups for each user
+    let mut groups_by_user = HashMap::new();
+    for user in &users {
+        // TODO: optimize by batching calls to OpenFGA
+        groups_by_user.insert(user.id, regulator.user_groups(&authz::User(user.id)).await?);
+    }
+    let group_ids: HashSet<i64> = groups_by_user.values().flatten().map(|g| g.0).collect();
+
+    let settings = SelectionSettings::new()
+        .filter(move || Group::ID.eq_any(group_ids.clone().into_iter().collect_vec()));
+    let groups = Group::list(&mut db_pool.get().await?, settings).await?;
+    let group_by_id = groups
+        .into_iter()
+        .map(|g| (g.id, g))
+        .collect::<HashMap<_, _>>();
+
+    // Order users in the same order as requested
+    let user_by_id: HashMap<_, _> = users.iter().map(|u| (u.id, u)).collect();
+    let user_by_identity: HashMap<_, _> = users.iter().map(|u| (&u.identity_id, u)).collect();
+    let ordered_users = body
+        .into_iter()
+        .map(|id| match id {
+            UserIdOrIdentity::Id(id) => user_by_id.get(&id).ok_or(AuthzError::UnknownUser { id }),
+            UserIdOrIdentity::Identity(identity) => user_by_identity
+                .get(&identity)
+                .ok_or(AuthzError::UnknownIdentity { identity }),
+        })
+        .collect::<Result<Vec<_>, _>>()?;
```

Comment:
Is this necessary? Why not just return an unordered map of user id => user info?